### DISCUSSION TITLE
Wrapper for RQ pagination and TS typings problem

### DISCUSSION THREAD

**Discussion by rlesniak** [TanStack/query#4495] [Q&A]
I created some simple wrapper for react-query to be able to easier manipulate pagination logic. But TS typings arent working as expected. Could anyone help me to properly infer `data` types?

Demo here: https://stackblitz.com/edit/vitejs-vite-p6zvsw?file=src/useGetSomething.ts&view=editor

**Comment by TkDodo** [comment]
the way you passed the generics to `UseQueryOptions` was off, here is a "fixed" version:

https://stackblitz.com/edit/vitejs-vite-jtibs7

"fixed" in quotes because it takes away your possibility to use the `select` option, as both generics (`TQueryFnData` and `TData`) are now the same. You would need _another_ generic like we have it internally to make that work again.

**Reply by rlesniak** [reply]
Thanks for response. TBH, I wanted to use `select` in some cases 😅 Is there any chance to fix it with working `select`?

**Reply by TkDodo** [reply]
yes, you need another generic :)

**Reply by rlesniak** [reply]
Can you please suggest what generic it should be? :( I have no idea :/
